library/Notifications/Model/Event.php

```diff
@@ -153,4 +155,112 @@ public static function mapSeverity(?string $severity): ?string
+            case 'downtime-end':
+                return t('left a downtime period', 'notifications.type');
+            case 'downtime-removed':
+                return t('prematurely left a downtime period', 'notifications.type');
+            case 'flapping-start':
+                return t('entered a flapping period', 'notifications.type');
+            case 'flapping-end':
+                return t('left a flapping period', 'notifications.type');
+            case 'incident-age':
+                return t('exceeded a time constraint', 'notifications.type');
+            default: // custom
+                return '';
+        }
+    }
+
+    /**
+     * Get the icon for the event
+     *
+     * @return ?Icon
+     */
+    public function getIcon(): ?Icon
+    {
+        $icon = null;
+
+        if ($this->type === 'state') {
+            $severity = $this->severity;
+            $class = 'severity-' . $severity;
+            switch ($severity) {
+                case 'ok':
+                    $icon = (new Icon(Icons::SEVERITY_OK, ['class' => $class]))->setStyle('fa-regular');
+                    break;
+                case 'crit':
+                    $icon = new Icon(Icons::SEVERITY_CRIT, ['class' => $class]);
+                    break;
+                case 'warning':
+                    $icon = new Icon(Icons::SEVERITY_WARN, ['class' => $class]);
+                    break;
+                case 'err':
+                    $icon = (new Icon(Icons::SEVERITY_ERR, ['class' => $class]))->setStyle('fa-regular');
+                    break;
+                case 'debug':
+                    $icon = new Icon(Icons::SEVERITY_DEBUG);
+                    break;
+                case 'info':
+                    $icon = new Icon(Icons::SEVERITY_INFO);
+                    break;
+                case 'alert':
+                    $icon = new Icon(Icons::SEVERITY_ALERT);
+                    break;
+                case 'emerg':
+                    $icon = new Icon(Icons::SEVERITY_EMERG);
+                    break;
+                case 'notice':
+                    $icon = new Icon(Icons::SEVERITY_NOTICE);
+                    break;
+            }
+
+            return $icon;
+        }
+
+        switch ($this->type) {
+            case 'acknowledgement-set':
+                $icon = new Icon(Icons::ACKNOWLEDGED);
+                break;
+            case 'acknowledgement-cleared':
+                $icon = new Icon(Icons::UNACKNOWLEDGED);
+                break;
+            case 'downtime-start':
+            case 'downtime-end':
+            case 'downtime-removed':
+                $icon = new Icon(Icons::DOWNTIME);
+                break;
+            case 'flapping-start':
+            case 'flapping-end':
+                $icon = new Icon(Icons::FLAPPING);
+                break;
+            case 'incident-age':
+                $icon = new Icon(Icons::INCIDENT_AGE);
+                break;
+            default:
```

Comment:
Please add a case here. I don't want this icon to be the default for any upcoming types.